### PR TITLE
Add missing member handling, deserialize string enum values

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #37101

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/standard/serialization/system-text-json/immutability.md](https://github.com/dotnet/docs/blob/fc48af7ad39921594a967b08945a7c82a20a72a0/docs/standard/serialization/system-text-json/immutability.md) | [Use immutable types and non-public members or accessors](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/immutability?branch=pr-en-us-37554) |
| [docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md](https://github.com/dotnet/docs/blob/fc48af7ad39921594a967b08945a7c82a20a72a0/docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md) | ["Migrate from Newtonsoft.Json to System.Text.Json - .NET"](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/migrate-from-newtonsoft?branch=pr-en-us-37554) |

### CODE REVIEW

File: docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md [hunk 1]
```diff
@@ -597,9 +624,15 @@ System.Text.Json doesn't have built-in support for these attributes. However, st
...
+
+* Apply the <xref:System.Text.Json.Serialization.JsonUnmappedMemberHandlingAttribute> attribute to the type you're deserializing to.
+* To set your preference globally, set the <xref:System.Text.Json.JsonSerializerOptions.UnmappedMemberHandling?displayProperty=nameWithType> property. Or, for source generation, set the <xref:System.Text.Json.Serialization.JsonSourceGenerationOptionsAttribute.UnmappedMemberHandling?displayProperty=nameWithType> property and apply the attribute to your <xref:System.Text.Json.Serialization.JsonSerializerContext> class.
+* Customize the <xref:System.Text.Json.Serialization.Metadata.JsonTypeInfo.UnmappedMemberHandling?displayProperty=nameWithType> property.
 
-Starting in .NET 7, you can use [contract customization](custom-contracts.md) as a workaround to match the `Newtonsoft.Json` functionality and throw an exception for JSON properties that don't exist in the target type. The following code snippet shows an example.
+In .NET 7, you can use [contract customization](custom-contracts.md) to throw an exception for JSON properties that don't exist in the target type. The following code snippet shows an example.
```
Comment (eiriktsarpalis) [inline docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md:635]: Given that the workaround is a bit of a hack, would it make sense to remove it altogether and just encourage folks to update?

**Review by eiriktsarpalis — APPROVED** [review]

File: docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md [hunk 2]
```diff
@@ -68,15 +68,16 @@ The following table lists `Newtonsoft.Json` features and `System.Text.Json` equi
...
+| Deserialize string enum value                         | ✔️ [Deserialize string enum values](#deserialize-string-enum-values) |
+| `MissingMemberHandling` global setting                | ✔️ [Handle missing members](#handle-missing-members) |
+| Add to collections without setters                    | ✔️ [](#add-to-collections-without-setters) |
 | Support for a broad range of types                    | ⚠️ [Some types require custom converters](#types-without-built-in-support) |
 | Deserialize inferred type to `object` properties      | ⚠️ [Not supported, workaround, sample](#deserialization-of-object-properties) |
 | Deserialize JSON `null` literal to non-nullable value types | ⚠️ [Not supported, workaround, sample](#deserialize-null-to-non-nullable-type) |
 | `DateTimeZoneHandling`, `DateFormatString` settings   | ⚠️ [Not supported, workaround, sample](#specify-date-format) |
 | `JsonConvert.PopulateObject` method                   | ⚠️ [Not supported, workaround](#populate-existing-objects) |
 | `ObjectCreationHandling` global setting               | ⚠️ [Not supported, workaround](#reuse-rather-than-replace-properties) |
```
Comment (eiriktsarpalis) [inline docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md:79]: We can remove this line as well, it's covered by the new "populate members" feature
```suggestion
```

File: docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md [hunk 3]
```diff
@@ -68,15 +68,16 @@ The following table lists `Newtonsoft.Json` features and `System.Text.Json` equi
 | `DefaultContractResolver` to ignore properties        | ✔️ [DefaultJsonTypeInfoResolver class](#conditionally-ignore-a-property) |
 | Polymorphic serialization                             | ✔️ [[JsonDerivedType] attribute](#polymorphic-serialization) |
 | Polymorphic deserialization                           | ✔️ [Type discriminator on [JsonDerivedType] attribute](#polymorphic-deserialization) |
+| Deserialize string enum value                         | ✔️ [Deserialize string enum values](#deserialize-string-enum-values) |
+| `MissingMemberHandling` global setting                | ✔️ [Handle missing members](#handle-missing-members) |
+| Add to collections without setters                    | ✔️ [](#add-to-collections-without-setters) |
```
Comment (eiriktsarpalis) [inline docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md:73]: We should probably generalize this beyond just collections
```suggestion
| Populate properties without setters                    | ✔️ [](#add-to-properties-without-setters) |
```

File: docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md (hunk 2 above)
Comment (gewarren) [reply docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md:79]: I will do the populate members work in a separate PR, and will remove this there.

File: docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md (hunk 3 above)
Comment (gewarren) [reply docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md:73]: Will address in separate PR.

**Pull request merged**